### ISSUE TITLE
Production guidelines

### ISSUE THREAD

**Issue by back-2-95** [chapter-three/next-drupal#94]
I tried to emulate creating "production" version of the Next.js part with Dockerfile and Docker Compose.

Was hoping to get some advices/examples on how to deploy next.js of this project to production.

My discoveries so far:

- I cannot use `npm ci --only=production` as `typescript` is in `devDependencies` of `package.json`
- I cannot use `npm run build` in Dockerfile as it tries to load content from Drupal
- Current CMD below works but fails quite fast as it cannot connect to Drupal container (Drupal is in `http://app:8080`) 

Here is the current Dockerfile:

```
ARG NODE_VERSION=16

FROM node:${NODE_VERSION}-alpine as deps

# Check https://github.com/nodejs/docker-node/tree/b4117f9333da4138b03a546ec926ef50a31506c3#nodealpine
# to understand why libc6-compat might be needed.
RUN apk add --no-cache libc6-compat

WORKDIR /app

COPY package.json package-lock.json /app/

#RUN npm ci --only=production
RUN npm ci

# Rebuild the source code only when needed
FROM node:${NODE_VERSION}-alpine as builder

ENV NEXT_TELEMETRY_DISABLED 1
ENV NODE_ENV production

WORKDIR /app

COPY . .
COPY --from=deps /app/node_modules ./node_modules
COPY .env.docker .env

RUN npm config set update-notifier false

ENV PORT 3000

EXPOSE 3000

CMD ["npm", "run", "preview"]
```

**Comment by shadcn** [comment]
Hmm, I don't use docker that often but I'll see if I can get someone on the team to help with this.

Does this example https://github.com/vercel/next.js/tree/canary/examples/with-docker help?

**Comment by back-2-95** [comment]
Maybe this needs to be thinked like:
- Production Drupal must be online
- Build prod image as part of any CI pipeline
- Dockerfile can on build time connect to production Drupal

**Comment by iamfredrik** [comment]
For me it was the TypeScript Interface NodePageProps etc. that was causing the problems. Removed them and it builds just fine.

**Comment by back-2-95** [comment]
I'd say now that we have launched our first next-drupal project I can say my initial thoughts in the issue is false.

I'm not saying one not could use Docker image BUT the main thing is that Drupal (the backend) must be available when that Docker image is build. Our initial problem was that hosting platform was building Drupal AND Next.js images at the same time (aka deployment) and next.js image build ofc failed.

What we did: we host next.js on Vercel and Drupal on Amazee.io Lagoon. Drupal instance(s) must be online and available for the Next.js builds.

Closing the issue.
